libs/blog-bff/shared/util-wp/src/lib/wp-client.ts

```diff
@@ -16,14 +16,31 @@ export class WPRestClient {
 
   get<T>(url: string, params: Record<string, string>): Promise<WPRespone<T>> {
     const searchParams = new URLSearchParams(params);
-    return this.request(`${url}?${searchParams.toString()}`, { method: 'GET' });
+    return this.request(`${url}?${searchParams.toString()}`, {
+      method: 'GET',
+    });
+  }
+
+  post<T>(name: string, body: T): Promise<WPRespone<T>> {
+    const form: FormData = new FormData();
+    form.append(name, body as string);
+
+    return this.request(
+      `subscribers`,
+      {
+        body: form,
+        method: 'POST',
+      },
+      '/wp-json/newsletter/v2/',
+    );
   }
 
   private async request<T>(
     url: string,
     { body, headers, method }: FetchConfig = {},
+    namespace = '/wp-json/wp/v2/',
```

Comment:
I feel like this does not belong here. You have two separate middlewares to handle different type of requests. Why not move the config to each middleware, so that `WPRestClient` stays generic?

Reply:
I find it easier to handle with. In fact, `WpRestClient` still stays generic, but with one additional parameter (which is not required). Newsletter's  `Post` method is an exception so I wouldn't bother too much about it. What is more, this solution was suggested earlier.